tests-e2e/core/snapshotenvironmentbinding_test.go

```diff
@@ -595,8 +596,8 @@ var _ = Describe("SnapshotEnvironmentBinding Reconciler E2E tests", func() {
 
 			err = k8s.Get(&gitopsDeployment, k8sClient)
 			Expect(err).To(BeNil())
-			Expect(gitopsDeployment.ObjectMeta.Labels).ToNot(BeNil())
-			Expect(gitopsDeployment.ObjectMeta.Labels["appstudio.openshift.io"]).To(Equal("testing"))
+			Eventually(gitopsDeployment.ObjectMeta.Labels, "2m", "10s").ShouldNot(BeNil())
+			Eventually(gitopsDeployment.ObjectMeta.Labels["appstudio.openshift.io"], "2m", "10s").Should(Equal("testing"))
```

Comment:
I believe neither of these `Eventually` calls are correct: the `gitopsDeployment` object will not be updated after the call to `k8s.Get` above.

Instead, you probably want to call the `HaveLabel` matcher from the `fixture/gitopsdeployment` package, here.

Reply:
Thanks for pointing to the utility methods defined in`fixture/gitopsdeployment`. I have corrected it now.